sql/src/main/java/io/crate/executor/transport/AlterTableOperation.java

```diff
@@ -114,7 +113,9 @@ public void onFailure(Throwable e) {
             return Futures.immediateFailedFuture(new AlterTableAliasException(table.ident().fqn()));
         }
 
-        List<ListenableFuture<Long>> results = new ArrayList<>(3);
+        final SettableFuture<Long> result = SettableFuture.create();
+        final AtomicInteger operations = new AtomicInteger(2);
```

Comment:
why change everything to the operations pattern instead of just adding the result of updateMapping/updateSettings to the results list where it was missing?